tiny_dnn/core/kernels/conv2d_op_internal.h

```diff
@@ -10,103 +10,114 @@
-         serial_size_t id          = params.in.depth_;
-         serial_size_t ow          = params.out.width_;
-         serial_size_t oh          = params.out.height_;
-         serial_size_t od          = params.out.depth_;
-         serial_size_t kw          = params.weight.width_;
-         serial_size_t kh          = params.weight.height_;
-         serial_size_t elem_stride = params.w_stride;
-         serial_size_t line_stride = iw * params.h_stride;
-         for (size_t sample = r.begin(); sample < r.end(); sample++) {
-           const vec_t &in = in_data[sample];
-           vec_t &a        = out_data[sample];
-           for (serial_size_t o = 0; o < od; o++) {
-             float_t *pa = &a[params.out.get_index(0, 0, o)];
-             for (serial_size_t inc = 0; inc < id; inc++) {
-               if (!params.tbl.is_connected(o, inc)) continue;
-               serial_size_t idx;
-               idx                = params.weight.get_index(0, 0, id * o + inc);
-               const float_t *pw  = &W[idx];
-               idx                = params.in_padded.get_index(0, 0, inc);
-               const float_t *pin = &in[idx];
-               float_t *pout      = pa;
-               for (serial_size_t y = 0; y < oh; y++) {
-                 const float_t *pin_line = pin;
-                 for (serial_size_t x = 0; x < ow; x++) {
-                   const float_t *pin_element = pin_line;
-                   const float_t *pw_element  = pw;
-                   float_t sum{0};
-                   // should be optimized for small kernel(3x3,5x5)
-                   for (serial_size_t wy = 0; wy < kh; wy++) {    // NOLINT
-                     for (serial_size_t wx = 0; wx < kw; wx++) {  // NOLINT
-                       sum += pw_element[wx] * pin_element[wx];
-                     }
-                     pw_element += kw;
-                     pin_element += iw;
-                   }
-                   pout[x] += sum;
-                   pin_line += elem_stride;
-                 }
-                 pout += ow;
-                 pin += line_stride;
-               }
-             }
-             if (params.has_bias) {
-               vectorize::add(bias[o], out_area, pa);
-             }
-           }
-         }
-       },
-       0);
+  for_(
+    parallelize, 0, in_data.shape()[0],
+    [&](const blocked_range &r) {
+      size_t out_area           = params.out.area();
+      serial_size_t iw          = params.in_padded.width_;
+      serial_size_t id          = params.in.depth_;
+      serial_size_t ow          = params.out.width_;
+      serial_size_t oh          = params.out.height_;
+      serial_size_t od          = params.out.depth_;
+      serial_size_t kw          = params.weight.width_;
+      serial_size_t kh          = params.weight.height_;
+      serial_size_t elem_stride = params.w_stride;
+      serial_size_t line_stride = iw * params.h_stride;
+      for (size_t sample = r.begin(); sample < r.end(); sample++) {
```

Comment:
it's okay for now but this can be simplified just just tensor accessors. I'll handle it later, just add a todo for me :smile: